Content.Shared/GameObjects/Components/Buckle/SharedBuckleComponent.cs

```diff
@@ -3,28 +3,51 @@
 using Content.Shared.GameObjects.EntitySystems;
 using Content.Shared.Interfaces.GameObjects.Components;
 using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Components;
 using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Physics;
 using Robust.Shared.Serialization;
 
 namespace Content.Shared.GameObjects.Components.Buckle
 {
-    public abstract class SharedBuckleComponent : Component, IActionBlocker, IEffectBlocker, IDraggable
+    public abstract class SharedBuckleComponent : Component, IActionBlocker, IEffectBlocker, IDraggable, ICollideSpecial
     {
         public sealed override string Name => "Buckle";
 
         public sealed override uint? NetID => ContentNetIDs.BUCKLE;
-
+       
         /// <summary>
         ///     True if the entity is buckled, false otherwise.
         /// </summary>
         public abstract bool Buckled { get; }
+        public  EntityUid? EntityBuckledTo { get; set; }
 
+        public  bool IsOnStrapEntityThisFrame { get; set; }
+        public  bool DontCollide { get; set; }
         public abstract bool TryBuckle(IEntity user, IEntity to);
 
+        protected IPhysicsComponent Body;
```

Comment:
Mark this with `ComponentDependency` and then remove `Owner.TryGetComponent(out Body);` in `Initialize()`, then it will get handled automatically for you.

Reply:
I don't know if I'm doing something wrong but when I try these changes, the server crashes with this error:
```
Robust.Shared.GameObjects.ComponentDependencies.ComponentDependencyNotNullableException: 'Field Robust.Shared.GameObjects.Components.IPhysicsComponent Body of Type Content.Server.GameObjects.Components.Buckle.BuckleComponent is marked as ComponentDependency, but does not have ?(Nullable)-Flag!'
```
I've tried setting Body as nullable but this gives me compiler errors.

Reply:
You are missing `#nullable enable` at the top of the file, but other than that the compiler errors after doing it would be because you need to check if its null for every case where you use it.

Reply:
Suggestion applied, thanks for the review btw.